libs/math/transform.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2015, Simon Fuhrmann, Nils Moehrle
+ * TU Darmstadt - Graphics, Capture and Massively Parallel Computing
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the BSD 3-Clause license. See the LICENSE.txt file for details.
+ */
+
+#ifndef MATH_TRANSFORM_HEADER
+#define MATH_TRANSFORM_HEADER
+
+#include "math/matrix.h"
+#include "math/matrix_svd.h"
+
+MATH_NAMESPACE_BEGIN
+
+template <typename T, int N>
+bool
+determine_transform(std::vector<math::Vector<T, N> > const& p0,
+    std::vector<math::Vector<T, N> > const& p1,
+    math::Matrix<T, N, N>* rot, T* scale, math::Vector<T, N>* trans)
+{
+    if (p0.size() != p1.size())
+        throw std::invalid_argument("Dimension size mismatch");
+
+    std::size_t num_correspondences = p0.size() & p1.size();
```

Comment:
Sorry habit both values are the same -> their binary representations is the same so the statement is will give the same result as ` std::size_t num_correspondences = p0.size();` but show the intent that both should be the same.